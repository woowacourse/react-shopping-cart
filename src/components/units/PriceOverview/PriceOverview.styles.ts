import styled from '@emotion/styled';

type RootProps = {
  border?: boolean;
};

const Root = styled.div<RootProps>`
  width: 448px;
  display: flex;
  flex-direction: column;
  border: ${(props) => (props.border ? `1px solid  ${props.theme.borderColor.lightGrey}` : 'none')};
`;

const TotalPriceHeader = styled.div`
  width: 100%;
  padding: 22px 30px;
  font-size: 24px;
`;

const Divider = styled.hr`
  width: 100%;
  height: 3px;
  margin: 0;
  background-color: ${({ theme }) => theme.borderColor.lightGrey};
  border: none;
  box-sizing: border-box;
`;

const TotalPriceContent = styled.div`
  padding: 34px 30px;
`;

const OrderButton = styled.button`
  border: none;
  background-color: ${({ theme }) => theme.bgColor.primary};
  color: ${({ theme }) => theme.textColor.defaultWhite};
`;

export default { Root, TotalPriceHeader, Divider, TotalPriceContent, OrderButton };
