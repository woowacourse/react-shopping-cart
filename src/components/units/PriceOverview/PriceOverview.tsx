import React from 'react';
import Styled from './PriceOverview.styles';

interface PriceOverviewProps {
  headerText: string;
  children: React.ReactNode;
  border?: boolean;
}

const PriceOverview = (props: PriceOverviewProps) => {
  const { headerText, children, border } = props;

  return (
    <Styled.Root border={border}>
      <Styled.TotalPriceHeader>{headerText}</Styled.TotalPriceHeader>
      <Styled.Divider />
      <Styled.TotalPriceContent>{children}</Styled.TotalPriceContent>
    </Styled.Root>
  );
};

PriceOverview.defaultProps = {
  border: true,
};

export default PriceOverview;
