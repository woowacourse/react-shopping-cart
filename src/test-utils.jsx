import React from 'react';
import { Provider } from 'react-redux';
import { Global, ThemeProvider } from '@emotion/react';
import { SnackbarProvider } from 'notistack';
import { render } from '@testing-library/react';
import { store } from './modules/store';
import { globalStyle, theme } from './App.styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <SnackbarProvider
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        maxSnack={3}
      >
        <Global styles={globalStyle} />
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </SnackbarProvider>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
