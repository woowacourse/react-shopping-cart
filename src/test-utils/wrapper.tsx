import { MemoryRouter } from "react-router-dom";
import { ToastProvider } from "../contexts/ToastContext";
import { CartProvider } from "../contexts/CartContext";
import { CouponProvider } from "../contexts/CouponContext";

const wrapper = ({ children }: React.PropsWithChildren) => (
  <MemoryRouter>
    <ToastProvider>
      <CartProvider>
        <CouponProvider>{children}</CouponProvider>
      </CartProvider>
    </ToastProvider>
  </MemoryRouter>
);

export default wrapper;
