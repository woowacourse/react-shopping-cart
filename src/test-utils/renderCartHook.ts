import { renderHook } from "@testing-library/react";
import useCart from "../hooks/useCart";
import useCartCalculations from "../hooks/useCartCalculations";
import wrapper from "./wrapper";

export const renderCartHook = () =>
  renderHook(
    () => {
      const cart = useCart();
      const calculations = useCartCalculations();
      return { ...cart, ...calculations };
    },
    { wrapper }
  );
