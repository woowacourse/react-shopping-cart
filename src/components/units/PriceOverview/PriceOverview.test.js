import React from 'react';
import { render } from '../../../test-utils';
import PriceOverview from './PriceOverview';

describe('PriceOverview 테스트', () => {
  test('matches snapshot', () => {
    const headerText = '장바구니 목록';
    const utils = render(<PriceOverview headerText={headerText} />);

    expect(utils.container).toMatchSnapshot();
  });
});
