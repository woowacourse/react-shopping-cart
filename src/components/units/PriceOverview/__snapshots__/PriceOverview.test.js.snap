// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceOverview 테스트 matches snapshot 1`] = `
<div>
  <div
    class="css-2zmbk8-Root ebu2xg54"
  >
    <div
      class="css-ggfh6p-TotalPriceHeader ebu2xg53"
    >
      장바구니 목록
    </div>
    <hr
      class="css-1nhg6ge-Divider ebu2xg52"
    />
    <div
      class="css-91dqby-TotalPriceContent ebu2xg51"
    />
  </div>
</div>
`;
